filament/src/FrameInfo.cpp

```diff
@@ -26,6 +26,17 @@ namespace filament {
 using namespace utils;
 using namespace details;
 
+namespace details {
+// this is to avoid a call to memmove
+template<class InputIterator, class OutputIterator>
+static inline
+void move_backward(InputIterator first, InputIterator last, OutputIterator result) {
```

Comment:
This is cool.  Consider moving to `utils/algorithm.h`.  Also, I would've called it `rotate_right` since std::algorithm has a rotate function (which is actually a left rotation)

Reply:
I would think that "rotate" would keep the value that exists and reinsert it? This is actually more like a insert_front({}).